tests/serverpod_test_server/test_integration/database_operations/persist/non_persist_field_test.dart

```diff
@@ -0,0 +1,145 @@
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:serverpod_test_server/test_util/test_serverpod.dart';
+import 'package:test/test.dart';
+
+void main() async {
```

Comment:
Suggestion: Can you add a test with an object that has a relation that has a relation, e.g. `A <-> B <-> C`.

When doing an insert with A when B and C is present in the object then both B and C should be returned in the final object.

Reply:
E.g. I would like to verify that nested objects that are not persistent also are taken into account.

Reply:
okay, on it

Reply:
done!